xlunch.c

```diff
@@ -1957,7 +1965,7 @@ void init(int argc, char **argv)
         exit(WINERROR);
     }
 
-    XMatchVisualInfo(disp, DefaultScreen(disp), 32, TrueColor, &vinfo);
+    XMatchVisualInfo(disp, DefaultScreen(disp), 24, TrueColor, &vinfo);
```

Comment:
Please change this back to support translucent backgrounds and background colours.

Reply:
We might need a switch or check here then, as it won't build on my system (or any other recent Puppy Linux) unless it's 24 ... Not sure what the check would be.. 

See here for more info: http://murga-linux.com/puppy/viewtopic.php?p=998515#998515

Reply:
Hmm, that's a shame. Is it right that it doesn't build? Or that it just crashes at runtime as that post seems to indicate? If it's the latter it seems like an easy check to see if `XMatchVisualInfo` returns non-nil and then try it again with 24.

Reply:
crashed at runtime...

Reply:
Okay, in that case it should just be a case of checking the return value of `XMatchVisualInfo`, if it's not 0 then it should be called again with 24 as  the depth. Could you try this and see if it works?

Reply:
I will try it when I get home

Reply:
Just fixed this in https://github.com/Tomas-M/xlunch/commit/85d4163a5eebdd5ba49e85cd80496d0a8fc24ceb, could you test that and see if it works for you?